test/extensions/filters/http/common/fuzz/decoder_filter_fuzz.h

```diff
@@ -0,0 +1,30 @@
+#include <chrono>
+#include <memory>
+
+#include "envoy/http/filter.h"
+
+#include "test/fuzz/common.pb.validate.h"
+#include "test/fuzz/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+
+// Base class for fuzzing a decoder filter.
+class DecoderFuzzFilter {
+public:
+  void fuzz(const test::fuzz::HttpData& data) {
+    Http::TestHeaderMapImpl headers = Fuzz::fromHeaders(data.headers());
+    // TODO: add end_stream handling and decodeTrailers
+    filter_->decodeHeaders(headers, false);
+    Buffer::OwnedImpl buffer(data.data());
+    filter_->decodeData(buffer, true);
```

Comment:
For `decodeData`, could we call it multiple times for the buffer, with the last call being `end_stream=true`? Some of our custom filters have logic to handle streaming grpc data, so a single call would not encapsulate this case.

Reply:
Yeah, that's a good point I asked about in the GH issue. We could make the inputs look more like an action stream with repeated Headers and Data, and call decode multiple times with the last call ending the stream. WDYT? It's useful to know that's good, I'll change at least the data to be repeated strings.